File: PR1/PR1_Ej3/bckp/HamburguesaSinGlutenBuilder.py
from bckp.HamburguesaBuilder import HamburguesaBuilder

class HamburguesaSinGlutenBuilder(HamburguesaBuilder):
    def __init__(self):
        self.ingredientes = []

    def construirPanSinGluten(self):  # Pan SIN gluten
        self.ingredientes.append("Pan SIN gluten")

    def construirCarne(self):  # Carne
        self.ingredientes.append("Carne")

    def construirTomate(self):  # Tomate
        self.ingredientes.append("Tomate")

    def construirLechuga(self):  # Lechuga
        self.ingredientes.append("Lechuga")

    def construirQueso(self):  # Queso
        self.ingredientes.append("Queso")

    def construirBacon(self):  # Bacon
        self.ingredientes.append("Bacon")

    def construirCebolla(self):  # Cebolla
        self.ingredientes.append("Cebolla")

    def construirHuevo(self):  # Huevo
        self.ingredientes.append("Huevo")

    def construirSalsa(self):  # Salsa
        self.ingredientes.append("Salsa")

File: PR2/pantalla_pedidos_hamburgueseria/lib/HamburguesaSinGlutenBuilder.dart
import 'HamburguesaBuilder.dart';
import 'Hamburguesa.dart';
import 'dart:async';

class HamburguesaSinGlutenBuilder implements HamburguesaBuilder {
  Hamburguesa hamburguesa = Hamburguesa();

  @override
  void aniadePan() {
    hamburguesa.pan = "Pan sin gluten";
    sleepLong();
  }

  @override
  void aniadeLechuga() {
    hamburguesa.lechuga = "Lechuga verde";
    sleep();
  }

  @override
  void aniadeTomate() {
    hamburguesa.tomate = "Tomate pera";
    sleep();
  }

  @override
  void aniadeQuesoCabra() {
    hamburguesa.quesoCabra = "Queso de cabra (sin trazas)";
    sleep();
  }

  @override
  void aniadeCebolla() {
    hamburguesa.cebolla = "Cebolla llorosa";
    sleep();
  }

  @override
  void aniadePepinillos() {
    hamburguesa.pepinillos = "Pepinillos";
    sleep();
  }

  @override
  void aniadeBacon() {
    hamburguesa.bacon = "Bacon grasiento";
    sleepLong();
  }

  @override
  void aniadeCarne() {
    hamburguesa.carne = "Carne Wagyu (sin trazas)";
    sleepLong();
  }

  @override
  void aniadePrecio() {
    hamburguesa.precio = 6;
  }

  void sleep() {
    // Simular una pausa de 0.1 segundos
    Future.delayed(Duration(milliseconds: 100));
  }

  void sleepLong() {
    // Simular una pausa de 0.8 segundos
    Future.delayed(Duration(milliseconds: 800));
  }
}